umbral-pre-python/src/lib.rs

```diff
@@ -42,6 +42,27 @@ fn from_bytes<T: HasSerializableBackend<U>, U: SerializableToArray>(bytes: &[u8]
         })
 }
 
+fn hash<T: HasSerializableBackend<U>, U: SerializableToArray>(obj: &T) -> PyResult<isize> {
+    let serialized = obj.as_backend().to_array();
+
+    // call `hash((class_name, bytes(obj)))`
+    Python::with_gil(|py| {
+        let builtins = PyModule::import(py, "builtins")?;
+        let arg1 = PyUnicode::new(py, T::name());
+        let arg2: PyObject = PyBytes::new(py, serialized.as_slice()).into();
+        builtins.getattr("hash")?.call1(((arg1, arg2),))?.extract()
+    })
+}
+
+// For some reason this lint is not recognized in Rust 1.46 (the one in CI)
+// remove when CI is updated to a newer Rust version.
```

Comment:
What about updating rust in the CI config?

Reply:
It is actually run on both Rust 1.46 (minimum supported Rust version, MSRV) and the current stable version. 1.46 was chosen because it's the MSRV of the Rust-Crypto stack. We could bump MSRV, but I wouldn't want to do that just because of a single lint.